kura/test/org.eclipse.kura.core.testutil/src/main/java/org/eclipse/kura/core/testutil/TestUtil.java

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Eurotech and/or its affiliates and others
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ ******************************************************************************/
+package org.eclipse.kura.core.testutil;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TestUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(TestUtil.class);
+
+    private static Field getField(Object svc, String fieldName) throws NoSuchFieldException {
+        Field field = null;
+        Class clazz = svc.getClass();
+        while (!(clazz == Object.class || field != null)) {
+            try {
+                field = clazz.getDeclaredField(fieldName);
+                break;
+            } catch (NoSuchFieldException e) {
+            }
+            clazz = clazz.getSuperclass();
+        }
+
+        if (field == null) {
+            throw new NoSuchFieldException("Field not found");
+        }
+
+        return field;
+    }
+
+    public static Object getFieldValue(Object svc, String fieldName) {
+        Object result = null;
+
+        try {
+            Field field = getField(svc, fieldName);
+            field.setAccessible(true);
+            result = field.get(svc);
+        } catch (IllegalArgumentException e) {
+            logger.warn(e.getMessage(), e);
+        } catch (IllegalAccessException e) {
+            logger.warn(e.getMessage(), e);
+        } catch (NoSuchFieldException e) {
+            logger.warn(e.getMessage(), e);
+        }
+
+        return result;
+    }
+
+    private static Method getMethod(Object svc, String methodName, Class... paramTypes) throws NoSuchMethodException {
+        Method method = null;
+        Class<?> clazz = svc.getClass();
+        hwile: while (!(clazz == Object.class || method != null)) {
```

Comment:
You will understand my reasons with the next pull request when this method is extended. It is not enough to simply return the method there. Parameters will be checked after a potential method is found.
And as far as I know, Kura is Java 6.